File: java/PlanesAndroid/app/src/main/java/com/planes/android/TwoLineTextButtonWithState.java
package com.planes.android;

import android.content.Context;
import android.util.AttributeSet;

public class TwoLineTextButtonWithState extends TwoLineTextButton implements ObjectWithStringState, ViewWithText {

    public TwoLineTextButtonWithState(Context context) {
        super(context);
        init();
    }

    public TwoLineTextButtonWithState(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public TwoLineTextButtonWithState(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init();
    }

    public void setState(String stateName) {
        //TODO: read texts from resources
        m_State = stateName;
        if (m_State == "player") {
            m_Text2 = "Player Board";
        }
        if (m_State == "computer") {
            m_Text2 = "Computer Board";
        }

        invalidate();
    }
    public String getCurrentStateName() {
        return m_State;
    }
    private String m_State;
}

File: kotlin/PlanesAndroid/app/src/main/java/com/planes_multiplayer/android/TwoLineTextButtonWithState.kt
package com.planes_multiplayer.android

import android.content.Context
import android.util.AttributeSet

class TwoLineTextButtonWithState : TwoLineTextButton, ObjectWithStringState, ViewWithText {
    constructor(context: Context?) : super(context) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle) {
        init()
    }

    override fun setState(stateName: String, text: String?) {
        //TODO: read texts from resources
        currentStateName = stateName
        if (currentStateName === "player") {
            m_Text2 = text
        }
        if (currentStateName === "computer") {
            m_Text2 = text
        }
        invalidate()
    }

    override var currentStateName: String? = null
}